### ISSUE TITLE
 touch -m -h -d 2009-10-10 link is failing with "Unable to parse date"

### ISSUE THREAD

**Issue by sylvestre** [uutils/coreutils#2311]
With
```
touch a 
ln -s a link
```
```
touch -m -h -d 2009-10-10 link
```
works with gnu
but fails on us with
```
$ ./target/release/coreutils touch -m -h -d 2009-10-10 link
thread 'main' panicked at 'Unable to parse date
Invalid day.', src/uu/touch/src/touch.rs:243:19
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Code is in:
https://github.com/uutils/coreutils/blob/master/src/uu/touch/src/touch.rs#L243

**Comment by SuperSandro2000** [comment]
Encountered a similar problem in https://github.com/NixOS/nixpkgs/pull/116274

```
touch --date '2021-05-30 16:59:52.627421773 +0000' /nix/store/hf1cjsvzyv8mfazndnycln2zy57hqfbz-ed-1.17/bin/red
thread 'main' panicked at 'Unable to parse date
Invalid day.', src/uu/touch/src/touch.rs:248:19
stack backtrace:
   0: rust_begin_unwind
   1: std::panicking::begin_panic_fmt
   2: uu_touch::parse_date
   3: uu_touch::uumain
   4: coreutils::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by siraben** [comment]
Is this still an issue?

**Comment by SuperSandro2000** [comment]
This was fixed in #2317
